libsolidity/analysis/TypeChecker.cpp

```diff
@@ -41,15 +41,13 @@ bool typeSupportedByOldABIEncoder(Type const& _type)
 {
 	if (_type.dataStoredIn(DataLocation::Storage))
 		return true;
-	else if (_type.category() == Type::Category::Struct)
+	if (_type.category() == Type::Category::Struct)
 		return false;
-	else if (_type.category() == Type::Category::Array)
+	if (_type.category() == Type::Category::Array)
 	{
 		auto const& arrayType = dynamic_cast<ArrayType const&>(_type);
 		auto base = arrayType.baseType();
-		if (!typeSupportedByOldABIEncoder(*base))
-			return false;
-		else if (base->category() == Type::Category::Array && base->isDynamicallySized())
+		if ((!typeSupportedByOldABIEncoder(*base))||(base->category() == Type::Category::Array && base->isDynamicallySized()))
```

Comment:
Please also adjust your changes such that they follow the coding guidelines: https://raw.githubusercontent.com/ethereum/solidity/develop/CODING_STYLE.md.

Reply:
@bit-shift Coding standard fixed. :slightly_smiling_face: